packages/flutter/lib/src/widgets/image.dart

```diff
@@ -215,7 +239,15 @@ class Image extends StatefulWidget {
     this.gaplessPlayback = false,
     this.filterQuality = FilterQuality.low,
     Map<String, String> headers,
-  }) : image = NetworkImage(src, scale: scale, headers: headers),
+    bool resizeToFit = false,
+  }) : image = _resizeIfNeeded(
```

Comment:
why do we do this for network image but not asset image or memory image?

Reply:
@Hixie , i wanted to get feedback on this `_resizeIfNeeded` pattern before i go ahead and change the other ones.

Reply:
I don't really understand the pattern as written. It seems very network-specific. I'm having trouble understanding how it generalizes.